editor/src/components/canvas/controls/grid-controls.tsx

```diff
@@ -2254,128 +2300,344 @@ const RulerMarkers = React.memo((props: { path: ElementPath }) => {
         height: height,
       })
 
+      let otherColumnMarkers: Array<RulerMarkerPositionData> = []
```

Comment:
I think this would become a lot simpler if you would just collect all the notch positions independently as canvas points. For example for the horizontal axis, you would just need to grab all the columns from the grid frames (`parentGridCellGlobalFrames[0]`), and concat them with one extra entry for the final notch with `x: lastCol.x + lastCol.width`. Then, you can just:
```ts
notches.map(notch => <div key={...} style={{ x: notch.x, y: grid.y - notchHeight, ... }} />)
```
(if you don't want to display the triangle marker on top of the notch you can just filter it out, but I believe the ruler should display _all_ the notches corresponding to the grid lines)

Reply:
Ah yes, I did start thinking about this, but then because of the grid gaps it didn't seem to make sense at the time.

Reply:
It also needed to have the selected start and end values called out specifically as some of the existing logic relies on those.